src/main/java/com/direwolf20/buildinggadgets/common/tools/NetworkExtractor.java

```diff
@@ -0,0 +1,69 @@
+package com.direwolf20.buildinggadgets.common.tools;
+
+import java.util.Collection;
+import java.util.List;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import com.google.common.collect.ImmutableList;
+import com.raoulvdberge.refinedstorage.api.network.INetwork;
+import com.raoulvdberge.refinedstorage.api.util.Action;
+
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.items.IItemHandler;
+
+public abstract class NetworkExtractor implements IItemHandler {
+    private final List<ItemStack> stacks;
+
+    protected NetworkExtractor(Collection<ItemStack> stacks) {
+        this.stacks = stacks.stream().collect(ImmutableList.toImmutableList());
```

Comment:
As you know don't need to filter it, this could be with .copyOf...

Reply:
Oh yeah, the standard way... whoops. :smile: